src/mlpack/tests/distribution_test.cpp

```diff
@@ -188,6 +188,29 @@ BOOST_AUTO_TEST_CASE(MultiDiscreteDistributionTrainProTest)
   BOOST_REQUIRE_CLOSE(d.Probability("2 1 0"), 0.015625, 1e-5);
 }
 
+/**
+ * Test the LogProbability() function, for multiple points in the multivariate
+ * Discrete case.
+ */
+BOOST_AUTO_TEST_CASE(DiscreteLogProbabilityTest)
+{
+  // Same case as before.
+  DiscreteDistribution d("5 5");
+
+  arma::mat obs("0 2;"
+                "1 2;");
+
+  arma::vec logProb;
+
+
+  d.LogProbability(obs, logProb);
+
+  BOOST_REQUIRE_EQUAL(logProb.n_cols, 2);
```

Comment:
`logProb` is an `arma::vec`, so it will only have 1 column.  You should be able to use `n_elem` here instead of `n_cols`, and I think the test will work.